src/test/java/org/mule/service/scheduler/internal/CustomSchedulerShutdownTestCase.java

```diff
@@ -0,0 +1,125 @@
+import static org.mule.service.scheduler.internal.config.ContainerThreadPoolsConfig.loadThreadPoolsConfig;
+import static org.mule.test.allure.AllureConstants.SchedulerServiceFeature.SCHEDULER_SERVICE;
+import static org.mule.test.allure.AllureConstants.SchedulerServiceFeature.SchedulerServiceStory.SHUTDOWN;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.api.util.concurrent.Latch;
+import org.mule.service.scheduler.internal.config.ContainerThreadPoolsConfig;
+import org.mule.service.scheduler.internal.threads.SchedulerThreadPools;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+
+import java.util.ArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Story;
+
+@Feature(SCHEDULER_SERVICE)
+@Story(SHUTDOWN)
+public class CustomSchedulerShutdownTestCase extends AbstractMuleTestCase {
+
+  private ContainerThreadPoolsConfig threadPoolsConfig;
+  private SchedulerThreadPools service;
+
+  private final long prestarCallbackSleepTime = 0L;
+
+  @Before
+  public void before() throws MuleException {
+    threadPoolsConfig = loadThreadPoolsConfig();
+    service = new SchedulerThreadPools(SchedulerThreadPoolsTestCase.class.getName(), threadPoolsConfig) {
+
+      @Override
+      protected void prestartCallback(CountDownLatch prestartLatch) {
+        super.prestartCallback(prestartLatch);
+        try {
+          sleep(prestarCallbackSleepTime);
+        } catch (InterruptedException e) {
+          currentThread().interrupt();
+          throw new MuleRuntimeException(e);
+        }
+      }
+    };
+    service.start();
+  }
+
+  @After
+  public void after() throws MuleException, InterruptedException {
+    if (service == null) {
+      return;
+    }
+    for (Scheduler scheduler : new ArrayList<>(service.getSchedulers())) {
+      scheduler.stop();
+    }
+    service.stop();
+  }
+
+  @Test
+  public void shutdownDoesntDestroyThreadGroup() throws InterruptedException, ExecutionException {
+    final Scheduler customScheduler = service.createCustomScheduler(config().withMaxConcurrentTasks(2), 2, () -> 10L);
+
+    CountDownLatch latch = new Latch();
+    AtomicReference<Thread> executedThread = new AtomicReference<>();
+    final Future<Boolean> interrupted = customScheduler.submit(() -> {
+      executedThread.set(currentThread());
+      try {
+        latch.await();
+        return false;
+      } catch (InterruptedException e) {
+        currentThread().interrupt();
+        return true;
+      }
+    });
+
+    customScheduler.shutdown();
+
+    Thread.sleep(5000);
+    latch.countDown();
```

Comment:
Although it's the same, I think it should be better to use `release`
```suggestion
    latch.release();
```